src/app/components/generate/generate.component.html

```diff
@@ -1,12 +1,34 @@
 <h1>Generate</h1>
-<div>Name & Type</div>
 
-<input type="text" [(ngModel)]="name" placeholder="Name">
+<div>Name</div>
+<input type="text" [(ngModel)]="options.mandatoryArgs.name">
 
-<select [(ngModel)]="type">
-    <option *ngFor="let optionalType of optionalTypes" [value]="optionalType.id">
-      {{ optionalType.displayName }}
+<div>Type</div>
+<select [(ngModel)]="options.mandatoryArgs.type" (change)="changeType($event.target.value)">
+  <option
+    *ngFor="let optionalType of optionalTypes"
+    [value]="optionalType.id">
+     {{ optionalType.displayName }}
   </option>
 </select>
 
-<button (click)="generateAngularClass()" [disabled]="!name">Generate</button>
+<div *ngIf="options.mandatoryArgs.type == angularClass.component">
+  <label>
+    <input type="checkbox" [(ngModel)]="options.optionalFlags.inLineTemplate.isActive">
+    <span>in-line template</span>
+  </label>
+</div>
+<div *ngIf="options.mandatoryArgs.type == angularClass.component">
```

Comment:
we don't have to use `ngIf` twice. just use it once on a common parent

Reply:
@Thatkookooguy 
Thanks, I've looked into it, very cool.
I guess it's not necessary now as we only have 2 check-boxes in the same div.